src/blocks/scratch3_motion.js

```diff
@@ -138,6 +139,59 @@ Scratch3MotionBlocks.prototype.glide = function (args, util) {
     }
 };
 
+Scratch3MotionBlocks.prototype.ifOnEdgeBounce = function (args, util) {
+    var bounds = util.target.getBounds();
+    // Measure distance to edges.
+    // Values are positive when the sprite is far away,
+    // and clamped to zero when the sprite is beyond.
+    var stageWidth = this.runtime.constructor.STAGE_WIDTH;
+    var stageHeight = this.runtime.constructor.STAGE_HEIGHT;
+    var distLeft = Math.max(0, stageWidth / 2 + bounds.left);
```

Comment:
This will throw here if there's no renderer due to `getBounds()` returning `null`. Maybe the function should just have an `if (!bounds) { return; }` or similar near the top?